src/index.ts

```diff
@@ -111,84 +110,108 @@ async function initRabbitConn(rabbituri: string, options: MTROptions) {
+                    await menash.bind(queue.exchange.name, queuedeclared, queue.exchange.routingKey);
+                }
             }
-        }
-    }));
+        })
+    );
 }
 
+/**
+ * Initiate stream from last event
+ */
+async function initiateChangeStreamStartTime(collectionName: string) {
+    const latestEvent = await changeStreamTrackerModel(collectionName).findOne({}).sort({ timeStamp: -1 });
+    const latestEventTime = latestEvent
+        ? new Timestamp(1, latestEvent.timeStamp * millisecondConvertNumber)
+        : new Timestamp(1, new Date().getTime() / millisecondConvertNumber - eventTtl);
+
+    return latestEventTime;
+}
 
 /**
  * Initializes the mongo watcher, given the mongo data.
  * @param {MongoDataType}   mongoData   - the data about mongo for the connection.
  * @param {RabbitDataType}  rabbitData  - the data about rabbit connection
  * @param {MTROptions}      options     - contains the MTROptions.
+ * @param {Timestamp}       startTime   - optional starttime listener
  */
-async function initWatch(mongoData: MongoDataType, rabbitData: RabbitDataType, options: MTROptions) {    
+async function initWatch(mongoData: MongoDataType, rabbitData: RabbitDataType, options: MTROptions, startTime?: Timestamp) {
     // Select DB and Collection
     const connectionObject: IConnectionStringParameters = csParser.parse(mongoData.connectionString);
     const db = mongoConn.db(connectionObject.endpoint);
     const collection = db.collection(mongoData.collectionName);
 
-    // Define change stream
-    const pipeline = [{ $match: {'ns.db': connectionObject.endpoint, 'ns.coll': mongoData.collectionName } }];
-    const optionsStream : ChangeStreamOptions = {fullDocument: 'updateLookup'}
+    // Define change stream settings
+    const pipeline = [{ $match: { 'ns.db': connectionObject.endpoint, 'ns.coll': mongoData.collectionName } }];
+
+    const optionsStream: ChangeStreamOptions = { fullDocument: 'updateLookup' };
+    if (startTime) optionsStream.startAtOperationTime = startTime;
+
     const changeStream = collection.watch(pipeline, optionsStream);
 
     // start listen to changes
-    changeStream.on('change', (event: mongodb.ChangeEvent<Object>) => {
-        log(`got mongo change event:  ${event.operationType} in collection:${mongoData.collectionName}`, options);
-        rabbitData.queues.forEach(queue => formatMsg(queue, options, event, mongoData));
-    }).on('error', async(err) => {
-        console.log(`error in mongo`);
-        console.log(err);
-        mongoConnection(mongoData, rabbitData, options);
-    });
+    changeStream
+        .on('change', (event: ChangeEvent) => {
+            log(`got mongo change event:  ${event.operationType} in collection:${mongoData.collectionName}`, options);
+
+            // Update event stream d
+            changeStreamTrackerModel(mongoData.collectionName).create(
```

Comment:
Where is the object removed?

Reply:
when it's expired

Reply:
How does it make sure that it is not expired by the time the service is up?